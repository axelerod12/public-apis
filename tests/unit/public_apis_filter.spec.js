import { shallowMount } from '@vue/test-utils';
import flushPromises from 'flush-promises';
import FeldmPublicApisFilter from '@/components/public_apis/public_apis_filter.vue';

const mockCategoriesList = {
  data: {
    categories: [
      'Animals',
      'Anime',
    ],
  },
};

// Following lines tell Jest to mock any call to `http.get`
// and to return `mockApiList` instead
jest.mock('axios', () => ({
  get: () => Promise.resolve(mockCategoriesList),
}));

describe('FeldmPublicApisFilter', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(FeldmPublicApisFilter);
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a label', () => {
    const label = wrapper.find('label');

    expect(label.exists()).toBe(true);
  });

  it('renders a select', () => {
    const select = wrapper.find('select');

    expect(select.exists()).toBe(true);
  });

  it('uses a mocked axios HTTP call when created', async () => {
    // Axios promise is resolved immediately.
    await flushPromises();

    expect(wrapper.vm.categories.length).toBe(2);
  });

  it('sets a value', async () => {
    const select = wrapper.find('select');

    await select.setValue('Animals');

    expect(select.element.value).toBe('Animals');

    // Wait until $emits have been handled.
    await wrapper.vm.$nextTick();

    // Assert event has been emitted.
    expect(wrapper.emitted('input')).toBeTruthy();
  });
});
