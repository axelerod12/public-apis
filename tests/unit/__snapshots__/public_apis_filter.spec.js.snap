// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeldmPublicApisFilter renders correctly 1`] = `
<div
  class="flex items-center"
>
  <label
    class="text-sm font-medium text-gray-900 dark:text-gray-300"
    for="filter"
  >
    
    Category:
  
  </label>
   
  <select
    class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full mx-4 my-2 p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
    id="filter"
  >
    <option
      value=""
    >
      All
    </option>
     
    <option
      value="Animals"
    >
      
      Animals
    
    </option>
    <option
      value="Anime"
    >
      
      Anime
    
    </option>
  </select>
</div>
`;
